modules/cli/src/main/scala/scala/cli/launcher/LauncherCli.scala

```diff
@@ -70,6 +70,15 @@ object LauncherCli {
     sys.exit(exitCode)
   }
 
+  def scalaCliVersion(cliVersion: String, cliScalaVersion: Option[String]): String =
+    cliScalaVersion match {
+      case Some(v) => v
+      case None => // it should be updated after migratation to Scala 3 => (Version(cliVersion) <= Version(x)) Constants.defaultScala213Version
+        if (cliVersion == "nightly") Properties.versionNumberString
+        else if (Version(cliVersion) <= Version("0.1.2")) Constants.defaultScala212Version
+        else Properties.versionNumberString
+    }
```

Comment:
as per my other comment, I'd cut the `cliScalaVersion` param from here, smth like this:
```scala
def scalaCliVersion(cliVersion: String): String =
  if (cliVersion == "nightly") Properties.versionNumberString
  else if (Version(cliVersion) <= Version("0.1.2")) Constants.defaultScala212Version
  else Properties.versionNumberString
```